### PR TITLE
Update Box2D to the latest version

### PR DESCRIPTION
**Pull request by Gareth422**
I have disabled the testbed, as it requires GLUI, which doesn't have a port yet.

This is for the following Google Code-in task: https://codein.withgoogle.com/dashboard/task-instances/5869573593104384/

### CODE REVIEW

**Comment by Gareth422** [comment]
There's still an issue with the install part of the recipe, as it's looking in the wrong location to put the headers. I've tried to patch this, but I can't find where to put it.

File: dev-games/box2d/box2d-2.3.1.recipe [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libgl$secondaryArchSuffix
+	devel:libglu$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	cd Box2D
+	cmake .
```
Comment (waddlesplash) [inline dev-games/box2d/box2d-2.3.1.recipe]: Add `$cmakeDirArgs` on the end of the line here.

File: dev-games/box2d/box2d-2.3.1.recipe [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	make
```
Comment (waddlesplash) [inline dev-games/box2d/box2d-2.3.1.recipe]: And `$jobArgs` here.

File: dev-games/box2d/patches/box2d-2.3.1.patch [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
++++ b/Box2D/CMakeLists.txt
+@@ -26,7 +26,8 @@ if(BOX2D_BUILD_EXAMPLES)
+ 
+   # Testbed and dependencies.
+   find_package(OpenGL REQUIRED)
++if(NOT HAIKU) # Haiku has glut, so no need for freeglut. Disable GLUI because it's not ported.
+   add_subdirectory(freeglut)
+   add_subdirectory(glui)
++endif(NOT HAIKU)
+-  add_subdirectory(Testbed)
```
Comment (waddlesplash) [inline dev-games/box2d/patches/box2d-2.3.1.patch]: Don't delete this line; just put it in the NOT HAIKU section if needed. Also, indent this section properly.

File: dev-games/box2d/patches/box2d-2.3.1.patch [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+@@ -207,8 +207,8 @@ 
+ 	# install build system hooks for third-party apps
+ 	install(EXPORT Box2D-targets DESTINATION ${LIB_INSTALL_DIR}/Box2D)
+ 
+-        set (BOX2D_INCLUDE_DIR ${CMAKE_INSTALL_PREFIX}/include)
++        set (BOX2D_INCLUDE_DIR /boot/system/develop/headers )
+-        set (BOX2D_INCLUDE_DIRS ${BOX2D_INCLUDE_DIR} )
++        set (BOX2D_INCLUDE_DIRS /boot/system/develop/headers )
+-        set (BOX2D_LIBRARY_DIRS ${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR})
++        set (BOX2D_LIBRARY_DIRS /boot/system/develop/lib )
```
Comment (waddlesplash) [inline dev-games/box2d/patches/box2d-2.3.1.patch]: Put back the `${}` constants, but add in relative paths e.g. `${CMAKE_INSTALL_PREFIX}/develop/headers` for `BOX2D_INCLUDE_DIR` for instance.

File: dev-games/box2d/patches/box2d-2.3.1.patch [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+-set ( BOX2D_INCLUDE_DIR  "@BOX2D_INCLUDE_DIR@" )
++set ( BOX2D_INCLUDE_DIR  "/boot/system/develop/headers" )
+-set ( Box2D_INCLUDE_DIRS "@BOX2D_INCLUDE_DIRS@" ) # deprecated
++set ( Box2D_INCLUDE_DIRS "/boot/system/develop/headers" ) # deprecated
+-set ( BOX2D_INCLUDE_DIRS "@BOX2D_INCLUDE_DIRS@" )
++set ( BOX2D_INCLUDE_DIRS "/boot/system/develop/headers" )
+ set ( BOX2D_LIBRARY      "@BOX2D_LIBRARY@" )
+ set ( BOX2D_LIBRARIES    "@BOX2D_LIBRARIES@" )
+-set ( BOX2D_LIBRARY_DIRS "@BOX2D_LIBRARY_DIRS@" )
++set ( BOX2D_LIBRARY_DIRS "/boot/system/develop/lib" )
```
Comment (waddlesplash) [inline dev-games/box2d/patches/box2d-2.3.1.patch]: And revert all changes to this file, since the previous comment will resolve that.

**Comment by waddlesplash** [comment]
Please put a link to the GCI task this is for in the description, and once you've fixed all those please squash your commits down to one.

**Comment by Gareth422** [comment]
It now installs to the correct location, but I get a policy error.

**Comment by Gareth422** [comment]
Alright, it all looks good now.

File: dev-games/box2d/box2d-2.3.1.recipe [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+winner of the 2008 Independant Game Festival Grand Prize"
+HOMEPAGE="http://www.box2d.org"
+SOURCE_URI="https://github.com/erincatto/Box2D/archive/v2.3.1.zip"
+LICENSE="MIT"
+COPYRIGHT="2007-2015 Erin Catto"
+CHECKSUM_SHA256="67c451bbf6d9b0ceae80e90c971f963ec149c59fb6ac73d8ebbf2c58c918f423"
+REVISION="1"
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+SECONDARY_ARCHITECTURES="x86"
+SOURCE_DIR="Box2D-2.3.1"
```
Comment (waddlesplash) [inline dev-games/box2d/box2d-2.3.1.recipe]: This is all out of order. See https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering for proper formatting.

**Comment by waddlesplash** [comment]
Thanks!

**Pull request merged**